yatsm/algorithms/ccdc.py

```diff
@@ -220,6 +220,9 @@ def fit(self, X, Y, dates):
 
             while not self.monitoring and self.can_monitor:
                 self.train()
+                # Ensure all bands in Y are fit in case we can't monitor
+                # after training
+                self._update_model()
                 self.here += 1
 
```

Comment:
My bad -- I misidentified where this snippet was inserted when reviewing your initial changes. 

The `self._update_model` call shouldn't happen in each training iteration as it is happening now. The update call should be outside of the training while loop, but before the monitoring loop has a chance to start:

``` python
while not self.monitoring and self.can_monitor:
    self.train()
    self.here += 1

# Ensure all bands are fit in case we can't monitor
self._update_model()

while self.monitoring and self.can_monitor:
    ....
```